build_tools/circle/build_test_arm.sh

```diff
@@ -21,10 +21,27 @@ source build_tools/shared.sh
 
 sudo add-apt-repository --remove ppa:ubuntu-toolchain-r/test
 sudo apt-get update
-sudo apt-get install python3-virtualenv ccache
-python3 -m virtualenv --system-site-packages --python=python3 testenv
-source testenv/bin/activate
-pip install --upgrade pip
+
+# Setup conda environment
+
+MINICONDA_URL="https://github.com/conda-forge/miniforge/releases/latest/download/Mambaforge-Linux-aarch64.sh"
+
+# Install Miniconda
+wget $MINICONDA_URL -O miniconda.sh
+MINICONDA_PATH=$HOME/miniconda
+chmod +x miniconda.sh && ./miniconda.sh -b -p $MINICONDA_PATH
+export PATH=$MINICONDA_PATH/bin:$PATH
+conda update --yes conda
+
+# Create environment and install dependencies
+conda create -n testenv --yes python=3.7
+source activate testenv
+
+# Use the latest by default
+mamba install --verbose -y ccache numpy scipy cython pip
+pip install joblib threadpoolctl
+pip install $(get_dep pytest $PYTEST_VERSION) pytest-xdist
+
 setup_ccache
 python -m pip install $(get_dep cython $CYTHON_VERSION) \
                       $(get_dep joblib $JOBLIB_VERSION)
```

Comment:
I think that to simplify the configuration of this build, we could install all latest versions of all dependencies using conda forge in a single `mamba install` command.